### ISSUE TITLE
edit:listing:start-custom crashes when pressing Enter with no items

### ISSUE THREAD

**Issue by zzamboni** [elves/elvish#658]
Within `edit:-narrow-read`, if you type a string that leaves no options, and press Enter, Elvish crashes.

To reproduce:
```
edit:insert:binding[Alt-z] = { edit:-narrow-read { each [s]{ put [&content=$s &display=$s] } [one two] } [arg]{ edit:insert-at-dot $arg[content] } }
```
Then press Alt-z. If you type "foo" (for example) so that none of the options are left, and press Enter, the following happens:

https://gist.github.com/zzamboni/8584ad18fa1844063381ec4a846ade5b

**Comment by xiaq** [comment]
Apparently `edit:listing:start-custom` has the same issue; repurposing this issue.

**Comment by xiaq** [comment]
To reproduce, type the following in the minibuf:

```elvish
edit:listing:start-custom []
```

And then press Enter. Elvish crashes.

**Comment by xiaq** [comment]
Fixed by #892!
